collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_pgsched.py

```diff
@@ -0,0 +1,438 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Simon Dodsley (simon@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: purefa_pgsched
+short_description: Manage protection groups replication schedules on Pure Storage FlashArrays
+description:
+- Modify or delete protection groups replication schedules on Pure Storage FlashArrays.
+author:
+- Pure Storage Ansible Team (@sdodsley) <pure-ansible-team@purestorage.com>
+options:
+  name:
+    description:
+    - The name of the protection group.
+    type: str
+    required: true
+  state:
+    description:
+    - Define whether to set or delete the protection group schedule.
+    type: str
+    default: present
+    choices: [ absent, present ]
+  schedule:
+    description: Which schedule to change.
+    type: str
+    choices: ['replication', 'snapshot']
+    required: True
+  enabled:
+    description: Enable the schedule being configured.
+    type: bool
+    default True
+  replicate_at:
+    description:
+    - Specifies the preferred time, on the hour, at which to replicate the snapshots.
+    - Specifies the preferred time as HH:MM:SS, using 24-hour clock, at which to generate snapshots.
```

Comment:
One more question, why did you decide to make this one as **24-hour** instead of the **12-hour** format like the `blackout_start` and `blackout_end`?

Reply:
24-hour was used for some as Ansible will automatically convert this to seconds, so it makes the modules simpler. The 12-hour blackouts were used as the blackout times have to be on-the-hour numbers and the GUI uses that format.

Reply:
I need to remove that first description line as `replicate_at` is not limited to "on-the-hour" as it states

Reply:
@iblackman have you pulled the latest patch to this. I have no problem enabling and disabling the schedules

Reply:
You are totally right, I must have got an older version. It worked flawless @sdodsley.